src/amp/lib/experiment.ts

```diff
@@ -0,0 +1,150 @@
+interface ExperimentModel<VariantData> {
+    sticky: boolean;
+    consentNotificationId?: string;
+    variants: {
+        [variantName: string]: VariantData;
+    };
+}
+
+type AmpModel = ExperimentModel<number>;
+
+type StyledExperimentModel = ExperimentModel<VariantConfig>;
+
+interface VariantConfig {
+    proportion: number;
+    style: string;
+}
+
+interface ExperimentCollection<ExperimentObject> {
+    [experimentName: string]: ExperimentObject;
+}
+
+export type AmpModelCollection = ExperimentCollection<AmpModel>;
+
+export type StyledModelCollection = ExperimentCollection<StyledExperimentModel>;
+
+type StylesCollection = ExperimentCollection<{ [variantName: string]: string }>;
+
+const extractFromVariants = (
+    variants: { [variantName: string]: VariantConfig },
+    fieldName: keyof VariantConfig,
+): { [variantName: string]: string | number } => {
+    return Object.entries(variants).reduce((acc, [variantName, config]) => {
+        acc = { ...acc, ...{ [variantName]: config[fieldName] } };
+        return acc;
+    }, {});
+};
+
+const isValidAmpModelVariant = (variant: {
+    [variantName: string]: string | number;
+}): variant is { [variantName: string]: number } => {
+    return Object.values(variant).every(x => typeof x === 'number');
+};
```

Comment:
can just be an expression 
```suggestion
}): variant is { [variantName: string]: number } => 
   Object.values(variant).every(x => typeof x === 'number');
```